packages/gatsby-source-shopify/README.md

```diff
@@ -157,86 +173,96 @@ Not set by default. If set to a string (example `MyStore`) node names will be `a
-) {
-  edges {
-    node {
-      id
-      storefrontId
-      featuredImage {
-        id
-        altText
-        gatsbyImageData(width: 910, height: 910)
+query {
+  products: allShopifyProduct {
+    nodes {
+      featuredMedia {
+        preview {
+          image {
+            gatsbyImageData
+          }
+        }
       }
     }
   }
 }
 ```
 
-You could then display the image in your component like this:
-
-```jsx
-import { GatsbyImage } from "gatsby-plugin-image"
-
-function ProductListing(product) {
-  return (
-    <GatsbyImage
-      image={product.featuredImage.gatsbyImageData}
-      alt={product.featuredImage.altText}
-    />
-  )
+#### Product Media Previews
+This query is commonly used on product pages to display images for all media types.
+```graphql
+query {
+  products: allShopifyProduct {
+    nodes {
+      media {
+        preview {
+          image {
+            gatsbyImageData
+          }
+        }
+      }
+    }
+  }
 }
 ```
 
-<div id="use-runtime-images"></div>
-
-### Use runtime images
-
-If you get Shopify images at runtime that don't have the `gatsbyImageData` resolver, for example from the cart or Storefront API, you can use the `getShopifyImage` function to create an imagedata object to use with `<GatsbyImage>`.
-
-It expects an `image` object that contains the properties `width`, `height` and `originalSrc`, such as [a Storefront API `Image` object](https://shopify.dev/docs/storefront-api/reference/common-objects/image).
-
-```jsx
-import { GatsbyImage } from "gatsby-plugin-image"
-import { getShopifyImage } from "gatsby-source-shopify"
-
-function CartImage(storefrontProduct) {
-  // This is data from Storefront, not from Gatsby
-  const image = storefrontProduct.images.edges[0].node
-  const imageData = getShopifyImage({
-    image,
-    layout: "fixed",
-    width: 200,
-    height: 200,
-  })
+💡 Note: This query will return images for all media types including videos.
 
-  return <GatsbyImage image={imageData} alt={image.altText} />
+#### Product Media Previews and Videos
+This query is commonly used on product pages to display images alongside videos.
+```graphql
+query {
+  products: allShopifyProduct {
+    nodes {
+      media {
+        preview {
+          image {
+            gatsbyImageData
+          }
+        }
+        ... on ShopifyExternalVideo {
+          embeddedUrl
+          host
+        }
+        ... on ShopifyVideo {
+          sources {
+            format
+            height
+            url
+            width
+          }
+        }
+      }
+    }
+  }
 }
 ```
 
-<div id="download-images-up-front"></div>
+💡 Note: Previous versions of this plugin exposed the `images` field on products. Although it made the plugin easier to interact with, it made it impossible to add videos to your products. The new version of the plugin exposes the `media` field directly, allowing you to query for all of the images, videos and 3D renderings that Shopify supports.
```

Comment:
What do you think about adding an example of using a fragment to get images? I think it would look something like this IIRC


```graphql
query {
  products: allShopifyProduct {
    nodes {
      media {
        ... on MediaImage {
            # fields...................
        }
      }
    }
  }
}
```

Reply:
There are a couple of caveats to this approach. I thought this would work as well, but the problem became that any objects that aren't MediaImage (ShopifyVideo, ExternalVideo, Model3d) would return empty objects, with no way to filter them out other than client-side. Although the query would work, the preview image query is more robust, especially because if the user ran this query when there were only images on their Shopify site it would work fine, but after anyone added a video to Shopify it would break the site with undefined errors.

My opinion would be to put that query in the migration guide, but if you have other considerations I haven't thought of I am open to hearing them!